### ISSUE TITLE
changing timezones only works partially on linux 24.04

### ISSUE THREAD

**Issue by emaayan** [cockpit-project/cockpit#21336]
i don't know exactly if this is a bug report or not.. but we've noticed that in ubuntu 24 changing the time zone doesn't seem to take effect in all places 
for example doing timedatectl set-timezone don't seem to touch /etc/timezone which is what java takes it's values from by default in ubuntu it seems that dpkg-reconfigure tzdata is considered the standard way of doing things but i understand you guys are using dbus and systemd api's which means this wont' get affected.

**Comment by emaayan** [comment]
what do you mean remove it? try to completely remove /etc/timezone and where java will try and take timezone info? won't that brake other stuff in ubuntu?

**Comment by martinpitt** [comment]
I'm no expert here, but Debian plans to remove /etc/timezone, see https://bugs.debian.org/822733. It was restored later on in https://tracker.debian.org/news/1421751/accepted-tzdata-2022g-6-source-into-unstable/ and the migration postponed to Debian 13. Hence I suggested trying what happens if you move it away now, if Java falls back to /etc/localtime. Fedora and RHEL haven't had /etc/timezone for many many years, and Java is definitively a thing for RHEL. So there's a good chance that it actually works.

We will not change anything here in Cockpit -- either /etc/timezone is obsolete, and then the distributions will clean it up eventually (that seems to be the current plan); or it isn't, then that systemd patch should be restored.

**Comment by emaayan** [comment]
thanks, will try that , i have some back plan to try this as well, but i'm
not sure if there are any "hooks" scripts that are triggered when someone
changes a timezone, plus i have to figure why there are conflicts between
cockpit-bridge and cockpit-pcp too (according to apt at least, but that's
another story)

timedatectl show --property=Timezone --value |  sudo tee /etc/timezone
>/dev/null
sudo dpkg-reconfigure --frontend noninteractive tzdata

**Comment by martinpitt** [comment]
cockpit-pcp is obsolete, an upgrade should remove it (it got absorbed into cockpit-bridge)

**Comment by emaayan** [comment]
> I'm no expert here, but Debian plans to remove /etc/timezone, see [bugs.debian.org/822733](https://bugs.debian.org/822733). It was restored later on in [tracker.debian.org/news/1421751/accepted-tzdata-2022g-6-source-into-unstable](https://tracker.debian.org/news/1421751/accepted-tzdata-2022g-6-source-into-unstable/) and the migration postponed to Debian 13. Hence I suggested trying what happens if you move it away now, if Java falls back to /etc/localtime. Fedora and RHEL haven't had /etc/timezone for many many years, and Java is definitively a thing for RHEL. So there's a good chance that it actually works.
> 
> We will not change anything here in Cockpit -- either /etc/timezone is obsolete, and then the distributions will clean it up eventually (that seems to be the current plan); or it isn't, then that systemd patch should be restored.

that seems to work too, but won't updating tzdata will automatically create this file? or do i need to remove tzdata too?

**Comment by martinpitt** [comment]
> won't updating tzdata will automatically create this file

Right, this was just a test, not a permanent solution for Ubuntu 22.04. I was mostly wondering if Java still needs /etc/timezone somehow (which would be a surprise, but you never know..)

> do i need to remove tzdata too?

Ugh no, this is very important. It's just /etc/timezone which is obsolete; /etc/localtime symlink and the actual time zone data continue to be (very) relevant.

Thanks for checking!

**Comment by martinpitt** [comment]
> not a permanent solution for Ubuntu 22.04

Sorry, you said 24.04, but same difference really -- Debian 13 will drop it, and some Ubuntu 25.XX or 26.XX will eventually inherit it.
